chirp/drivers/radtel_common.py

```diff
@@ -0,0 +1,264 @@
+from logging import Logger
+import struct
+from typing import Literal, Tuple, Protocol, Any, Callable
+
+from chirp import chirp_common, errors, util
+
+COMMAND_ACCEPT = b"\x06"
+CHARSET_HEX = "0123456789ABCDEFabcdef"
+BLOCK_SIZE = 0x10  # 16 bytes
+
+
+class RadtelLikeRadio(Protocol):
+    _fingerprint: bytes
+    _magic: bytes
+    _upper: int
+    pipe: Any
+    status_fn: Callable[[chirp_common.Status], None]
+
+    MEM_ROWS: int
+
+    def get_mmap(self) -> bytearray: ...
+
+
+def rt_clean_buffer(radio: RadtelLikeRadio, LOG: Logger):
+    radio.pipe.timeout = 0.005
+    junk = radio.pipe.read(256)
+    radio.pipe.timeout = 5  # 5000ms
```

Comment:
Please don't do this sort of thing. On Windows, setting the timeout value flushes the buffer and can do other things to the serial driver that don't happen on the other platforms. Switching the timeout in quick succession will give platform-dependent results and should be avoided. I imagine `Serial.reset_input_buffer()` is all you need.